stacks-signer/src/signerdb.rs

```diff
@@ -865,7 +866,19 @@ impl SignerDb {
         {
             // Always use the oldest block as our tenure start timestamp
             tenure_start_timestamp = Some(block_info.proposed_time);
-            tenure_process_time_ms += block_info.validation_time_ms.unwrap_or(0);
+            let non_bootcode_contract_call_block = block_info.block.txs.iter().any(|tx| {
```

Comment:
I think this works, but instead of needing to loop through all of the transactions for every block, what if we instead did this upon inserting into the database? When we get the `BlockValidateOk` response, we have the block cost. At that point, we can just set the `validation_time_ms` to 0 if the cost is 0.

Reply:
Sure can do. Was doing optimizations in another branch but its ballooning so will make this improvement now!

Reply:
Done!